src/pipecat/extensions/ivr/ivr_navigator.py

```diff
@@ -0,0 +1,486 @@
+
+        # XML pattern aggregation
+        self._aggregator = PatternPairAggregator()
+        self._setup_xml_patterns()
+
+        # Register IVR events
+        self._register_event_handler("on_ivr_stuck")
+        self._register_event_handler("on_ivr_completed")
+
+    async def start_conversation(self):
+        """Switch to conversation mode with conversation prompt and VAD timing.
+
+        Call this method to explicitly switch to conversation mode with the
+        conversation prompt and VAD parameters. This is typically called from
+        an on_ivr_completed event handler.
+
+        Example::
+
+            @navigator.event_handler("on_ivr_completed")
+            async def on_completed(ivr_processor):
+                # Start conversation with the customer service rep
+                await ivr_processor.start_conversation()
+        """
+        if self._conversation_prompt is None:
+            logger.warning("Cannot start conversation mode: no conversation_prompt provided")
+            return
+
+        logger.info("Starting conversation mode")
+
+        # Switch to conversation prompt
+        messages = [{"role": "system", "content": self._conversation_prompt}]
+        llm_update_frame = LLMMessagesUpdateFrame(messages=messages)
+        await self.push_frame(llm_update_frame, FrameDirection.UPSTREAM)
+
+        # Update VAD parameters for conversation response timing
+        vad_params = VADParams(stop_secs=self._conversation_response_delay)
+        vad_update_frame = VADParamsUpdateFrame(params=vad_params)
+        await self.push_frame(vad_update_frame, FrameDirection.UPSTREAM)
+
+    def _setup_xml_patterns(self):
+        """Set up XML pattern detection and handlers."""
+        # Register DTMF pattern
+        self._aggregator.add_pattern_pair("dtmf", "<dtmf>", "</dtmf>", remove_match=True)
+        self._aggregator.on_pattern_match("dtmf", self._handle_dtmf_pattern)
+
+        # Register IVR pattern
+        self._aggregator.add_pattern_pair("ivr", "<ivr>", "</ivr>", remove_match=True)
+        self._aggregator.on_pattern_match("ivr", self._handle_ivr_pattern)
+
+    async def process_frame(self, frame: Frame, direction: FrameDirection):
+        """Process frames and aggregate XML tag content.
+
+        Args:
+            frame: The frame to process.
+            direction: The direction of frame flow in the pipeline.
+        """
+        await super().process_frame(frame, direction)
+
+        if isinstance(frame, StartFrame):
+            # Push the StartFrame right away
+            await self.push_frame(frame, direction)
+
+            # Update the context with the appropriate prompt based on the initial mode
+            if self._initial_mode == "conversation":
+                if self._conversation_prompt is None:
+                    raise ValueError(
+                        "conversation_prompt is required when initial_mode is 'conversation'"
+                    )
```

Comment:
maybe we could do this in the constructor as a predoncidton, this way when we process StartFrame we know we are good.